commands/create_kubeconfig.go

```diff
@@ -11,9 +10,12 @@ import (
 
 	"github.com/fatih/color"
 	"github.com/giantswarm/gsclientgen/models"
+	"github.com/giantswarm/kubeconfig"
 	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger/microloggertest"
+	"github.com/giantswarm/operatorkit/client/k8srestconfig"
 	"github.com/spf13/cobra"
-	yaml "gopkg.in/yaml.v2"
+	"k8s.io/client-go/kubernetes/fake"
```

Comment:
Hmm this is a bit tricky. We should avoid using the test libraries in gsctl isn't nice.

I think in the kubeconfig library we should relax the validation so the logger and k8s client are not required. As they are not used for `NewKubeConfigForRESTConfig`.

Reply:
Or what do you think that we could create another struct in `spec.go` and implement two methods as `NewKubeConfigForRestConfig`? or we could implement it as class method (don't know equivalent word for golang :( )

Reply:
Sorry about the delayed reply.

 I think we should simplify the interface. We only need the context and logger for the `ForApp` method because it fetches the secret and we should have logging there.


So it would look like this.

```
type Interface interface {
	// NewRESTConfigForApp returns a Kubernetes REST Config for the cluster configured
	// in the kubeconfig section of the app CR.
	NewRESTConfigForApp(ctx context.Context, app v1alpha1.App) (*rest.Config, error)
	// NewKubeConfigForRESTConfig returns a kubeConfig bytes for the given REST Config.
	NewKubeConfigForRESTConfig(config *rest.Config, clusterName, namespace string) ([]byte, error)
	// NewRESTConfigForKubeConfig returns a REST Config for the given KubeConfigValue.
	NewRESTConfigForKubeConfig(kubeConfig []byte) (*rest.Config, error)
}
```

I think that's simpler and much easier to use in gsctl.

For changing the interface in golang AFAIK you can't have 2 methods with the same name and signature. Its more restrictive than other languages.

WDYT?

Reply:
That looks good, but I'm not sure we can set the loose restriction on `k8sclient` and `logger` when the same `kubeconfig` instance actually could use other `ForApp` methods even though it shouldn't

Reply:
Best option I can see for now is doing the config checks in the `ForApp` method. Maybe with a `validate` function as more methods might need it in future.